src/Control/Monad/Eff/Ref.purs

```diff
@@ -1,35 +1,41 @@
--- | This module defines an effect and actions for working with
--- | global mutable variables.
+-- | This module defines an effect and actions for working with global mutable
+-- | variables.
 -- |
--- | _Note_: The `Control.Monad.ST` provides a _safe_ alternative
--- | to global mutable variables when mutation is restricted to a
--- | local scope.
+-- | _Note_: The `Control.Monad.ST` provides a _safe_ alternative to global
+-- | mutable variables when mutation is restricted to a local scope.
 module Control.Monad.Eff.Ref where
 
-import Prelude (Unit, unit)
-import Control.Monad.Eff (Eff())
+import Prelude ((<$>), bind, pure)
+import Control.Monad.Eff (Eff)
 
 -- | The effect associated with the use of global mutable variables.
 foreign import data REF :: !
 
--- | A value of type `Ref a` represents a mutable reference
--- | which holds a value of type `a`.
+-- | A value of type `Ref a` represents a mutable reference which holds a value
+-- | of type `a`.
 foreign import data Ref :: * -> *
 
 -- | Create a new mutable reference containing the specified value.
 foreign import newRef :: forall s r. s -> Eff (ref :: REF | r) (Ref s)
 
--- | Read the current value of a mutable reference
+-- | Read the current value of a mutable reference.
 foreign import readRef :: forall s r. Ref s -> Eff (ref :: REF | r) s
 
--- | Update the value of a mutable reference by applying a function
--- | to the current value.
-foreign import modifyRef' :: forall s b r. Ref s -> (s -> { state :: s, value :: b }) -> Eff (ref :: REF | r) b
+-- | A function passed to `modifyRef'`.
+type ModifyState s b = s -> { newValue :: s, returnValue :: b }
```

Comment:
This one I prefer to keep. It was originally inlined until I realized it's repeated in `ST` and potentially any consumer. I'm not happy with the name however.